fix(stress): fix c-s timeouts

soyacz opened the pull request:
Recently c-s changed behavior and for duration workflows: it added warmup iterations. For heavy load, warmup doesn't fit to 10 minutes duration margin we assumed for c-s startup and causes SCT to timeout c-s thread.

This fix removes warmup by appending ` no-warmup` to all c-s cmd's. 
refs: #5767

## PR pre-checks (self review)
<!--- PR should be created as Draft, when CI finished and relevant checkboxes selected, add reviewers and then click on "Ready for review" button.-->
<!--- Put an `x` in all the boxes that apply or create PR and then click on all relevant checkboxes: -->
- [ ] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [ ] I didn't leave commented-out/debugging code
- [ ] I added the relevant `backport` labels
- [ ] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [ ] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [ ] All new and existing unit tests passed (CI)
- [ ] I have updated the Readme/doc folder accordingly (if needed)

fruch commented on sdcm/tester.py:
```diff
@@ -1807,6 +1807,8 @@ def run_stress_cassandra_thread(
             timeout = get_timeout_from_stress_cmd(stress_cmd) or self.get_duration(duration)
 
+        if "no-warmup" not in stress_cmd:
+            stress_cmd += " no-warmup"
```
It's plain like that ? It shouldn't be part of a specific group of parameters ?

soyacz replied:
your'e right it's not, need to put it other place

soyacz replied:
fixed

fruch commented on sdcm/tester.py:
```diff
@@ -1807,6 +1807,11 @@ def run_stress_cassandra_thread(
         else:
             timeout = get_timeout_from_stress_cmd(stress_cmd) or self.get_duration(duration)
 
+        if "no-warmup" not in stress_cmd and "stress user" not in stress_cmd:
```
I think it's better to use `user profile=`
```suggestion
        if "no-warmup" not in stress_cmd and "user profile=" not in stress_cmd:
```

fruch replied:
Also we are sure user-profile doesn't do warm-up ?

soyacz replied:
damn, I tested it but didn't notice it was already set. It supports no-warmup.

soyacz replied:
fixed, and I found one more use case where before c-s was some java settings provided - also fixed.

fruch reviewed (dismissed):
LGTM

vponomaryov reviewed (approved):
LGTM

fgelcer commented on sdcm/tester.py:
```diff
@@ -1807,6 +1807,13 @@ def run_stress_cassandra_thread(
             timeout = get_timeout_from_stress_cmd(stress_cmd) or self.get_duration(duration)
 
+        if "no-warmup" not in stress_cmd:
+            # add no-warmup to stress_cmd if it's not there see issue #5872
```
did you mean issue #5767 ?

soyacz replied:
yes, fixed

fgelcer reviewed (dismissed):
LGTM

soyacz commented:
I switched the place - I noticed I did it in `run_stress_cassandra_thread` which is not best place as still some places bypass it. I moved it to `CassandraStressThread` and used `re.sub` for brevity.
Please re-review.

fgelcer reviewed (approved):
LGTM

vponomaryov reviewed (approved):
LGTM
